tests/ops/test_qubit_ops.py

```diff
@@ -941,6 +943,217 @@ def test_controlled_phase_shift_decomp(self, phi):
+
+        @qml.qnode(dev)
+        def circuit(phi):
+            qml.PauliX(wires=0)
+            qml.PauliX(wires=1)
+            excitation(phi, wires=[0, 1, 2, 3])
+
+            return qml.state()
+
+        assert np.allclose(state, circuit(np.pi / 2))
+
+    @pytest.mark.parametrize(("excitation", "phi"), [(qml.DoubleExcitation, -0.1),
+                                                     (qml.DoubleExcitationPlus, 0.2),
+                                                     (qml.DoubleExcitationMinus, np.pi / 4)])
+    def test_autograd_grad(self, excitation, phi):
+        """Tests that gradients are computed correctly using the
+        autograd interface"""
+
+        pytest.importorskip("autograd")
+
+        dev = qml.device('default.qubit.autograd', wires=4)
+
+        @qml.qnode(dev)
+        def circuit(phi):
+            qml.PauliX(wires=0)
+            qml.PauliX(wires=1)
+            excitation(phi, wires=[0, 1, 2, 3])
+
+            return qml.expval(qml.PauliZ(0))
+
+        assert np.allclose(qml.grad(circuit)(phi), np.sin(phi))
+
+    @pytest.mark.parametrize("diff_method", ["parameter-shift", "backprop"])
+    @pytest.mark.parametrize(("excitation", "phi"), [(qml.DoubleExcitation, -0.1),
+                                                     (qml.DoubleExcitationPlus, 0.2),
+                                                     (qml.DoubleExcitationMinus, np.pi / 4)])
+    def test_tf_grad(self, excitation, phi, diff_method):
+        """Tests that gradients are computed correctly using the
+        tensorflow interface"""
+
+        tf = pytest.importorskip("tensorflow")
+        dev = qml.device('default.qubit', wires=4)
+
+        @qml.qnode(dev, interface="tf", diff_method=diff_method)
+        def circuit(phi):
+            qml.PauliX(wires=0)
+            qml.PauliX(wires=1)
+            excitation(phi, wires=[0, 1, 2, 3])
+            return qml.expval(qml.PauliZ(0))
+
+        phi_t = tf.Variable(phi, dtype=tf.float64)
+        with tf.GradientTape() as tape:
+            res = circuit(phi_t)
+
+        grad = tape.gradient(res, phi_t)
+        print('grad=', grad)
+        print('analytical', np.sin(phi))
+        assert np.allclose(grad, np.sin(phi))
+
+    @pytest.mark.parametrize("diff_method", ["parameter-shift", "backprop"])
+    @pytest.mark.parametrize(("excitation", "phi"), [(qml.DoubleExcitation, -0.1),
+                                                     (qml.DoubleExcitationPlus, 0.2),
+                                                     (qml.DoubleExcitationMinus, np.pi / 4)])
+    def test_jax_grad(self, excitation, phi, diff_method):
+        """Tests that gradients and operations are computed correctly using the
+        jax interface"""
+
+        if diff_method=="parameter-shift":
+            pytest.skip("JAX support for the parameter-shift method is still TBD")
+
+        jax = pytest.importorskip("jax")
+        from jax import numpy as jnp
+
+        dev = qml.device('default.qubit', wires=4)
```

Comment:
Just for readability
```suggestion
        dev = qml.device('default.qubit.jax', wires=4)
```